modules/aws-static-website-configuration/cloudfront.tf

```diff
@@ -49,16 +50,16 @@ resource "aws_cloudfront_distribution" "this" {
   }
 
   dynamic "viewer_certificate" {
-    for_each = var.domain != "" ? [] : [0]
+    for_each = aws_acm_certificate.this
     content {
       cloudfront_default_certificate = true
     }
   }
 
   dynamic "viewer_certificate" {
-    for_each = var.domain != "" ? [0] : []
+    for_each = aws_acm_certificate.this
     content {
-      acm_certificate_arn = aws_acm_certificate.this[0].arn
+      acm_certificate_arn = each.value.arn
       ssl_support_method  = "sni-only"

```

Comment:
It should, since `for_each` here applies to `aws_acm_certificate.this`, which itself has `count = length(var.domains) > 0 ? 1 : 0`. If there aren't any resources (a resource with `count = 0` is a "noop"), it shouldn't be created and it's akin to `for_each = []`.

I could be wrong about this bit of Terraform behavior, though. This is another thing we should probably test out!